### PR TITLE
Bond yield other asset

### LINKED ISSUE #20: Stage 2 Bond Yield incorrect
The user adds two currencies but we only calculate the yield based on sEURO. So they are getting an incorrect yield, not reflecting the value they put in.

**Action point:** the yield of the bonding must be based on the value of the two currency pairs added, one of which is SEURO. Normalise the value of assets where appropriate.

### PR DESCRIPTION
**Pull request by ewansheldon**
resolves #20 
also removes sEURO test suite, as the behaviour of the sEURO token is now managed by [the stablecoins repo](https://github.com/the-standard/stablecoins)

### CODE REVIEW

File: contracts/StandardTokenGateway.sol [hunk 1]
```diff
@@ -96,10 +103,22 @@ contract StandardTokenGateway is AccessControl {
...
 
+    function seuroToStandardToken(uint256 _seuroAmount) public view returns (uint256) {
+        return inversed ?
+            _seuroAmount * tokenPrice :
+            _seuroAmount / tokenPrice;
+    }
+
+    function otherToStandardToken(uint256 _otherAmount) external view returns (uint256) {
+        (,int256 eurOtherRate,,,) = IChainlink(chainlinkEurOther).latestRoundData();
+        uint256 seuroValue = _otherAmount * 10 ** chainlinkDec / uint256(eurOtherRate);
```
Comment (therealdannzor) [inline contracts/StandardTokenGateway.sol:118]: I was thinking we make the gateway a hub for rewards. So we let many different reward bonds (i.e., type and decimals) interact with the gateway. Hence, the CL decimals must be flexible and we ideally don't re-deploy it unless critical.

Can you parametrise the decimals instead?

File: test/standardTokenGateway.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  beforeEach(async () => {
+    const ChainlinkMock = await (await ethers.getContractFactory('Chainlink')).deploy(eurUsdPrice);
+    const TST = await (await ethers.getContractFactory('DUMMY')).deploy('Standard Token', 'TST', 18);
+    const SEuro = await (await ethers.getContractFactory('DUMMY')).deploy('SEuro', 'SEUR', 18);
+    StandardTokenGateway = await (await ethers.getContractFactory('StandardTokenGateway')).deploy(
+      TST.address, SEuro.address, ChainlinkMock.address, CHAINLINK_DEC
+    );
+  });
+
+  it('provides the TST exchange for sEURO and the other asset', async () => {
```
Comment (therealdannzor) [inline test/standardTokenGateway.js:19]: good to create a new test file for gateway, can you add one with 6 decimals too (e.g, USDT)?

**Review by therealdannzor — CHANGES_REQUESTED** [review]
Great!

File: contracts/StandardTokenGateway.sol (hunk 1 above)
Comment (ewansheldon) [reply contracts/StandardTokenGateway.sol:118]: ok yeah i think i understand what you mean

File: test/standardTokenGateway.js (hunk 2 above)
Comment (ewansheldon) [reply test/standardTokenGateway.js:19]: this test suite probably won't exist if the calculations are moved to outside of this gateway contract

Comment (therealdannzor) [reply test/standardTokenGateway.js:19]: that's alright, just so we test with other decimal values than the erc-20 default one (18)

**Review by therealdannzor — APPROVED** [review]
LGTM 👍

**Pull request merged**